patches/ninja-1.11.1.1-wrap-system-ninja.patch

```diff
@@ -0,0 +1,255 @@
+Quick hack to have the ninja wheel wrap the binary from the ninja-build
+RPM rather than including a binary in the weel.
+
+Probably plenty more to investigate here - e.g. this can now be an
+arch-independent wheel.
+
+ninja_syntax.py comes from the misc/ dir in
+https://github.com/Kitware/ninja/archive/v1.11.1.g95dee.kitware.jobserver-1.tar.gz
+
+Simple test that this works ...
+
+Upstream wheel:
+
+ $ ninja --version
+ 1.11.1.git.kitware.jobserver-1
+
+With this patch:
+
+ $ rpm -q ninja-build
+ ninja-build-1.10.2-6.el9.x86_64
+ $ /usr/bin/ninja --version
+ 1.10.2
+ $ which ninja
+ .../work-dir/venv-python3.9/bin/ninja
+ $ ninja --version
+ 1.10.2
+
+diff -urNp ninja-1.11.1.1.orig/setup.py ninja-1.11.1.1/setup.py
+--- ninja-1.11.1.1.orig/setup.py	2023-10-09 18:05:36.000000000 +0100
++++ ninja-1.11.1.1/setup.py	2024-04-05 18:06:46.740928096 +0100
+@@ -5,7 +5,8 @@ import os
+ import sys
+ from distutils.text_file import TextFile
+ 
+-from skbuild import setup
++#from skbuild import setup
++from setuptools import setup
+ 
+ # Add current folder to path
+ # This is required to import versioneer in an isolated pip build
+diff -urNp ninja-1.11.1.1.orig/src/ninja/__init__.py ninja-1.11.1.1/src/ninja/__init__.py
+--- ninja-1.11.1.1.orig/src/ninja/__init__.py	2023-10-09 18:05:36.000000000 +0100
++++ ninja-1.11.1.1/src/ninja/__init__.py	2024-04-05 18:07:17.047746482 +0100
+@@ -44,7 +44,7 @@ if not os.path.exists(DATA):
+     if os.path.exists(_data):
+         DATA = _data
+ 
+-BIN_DIR = os.path.join(DATA, 'bin')
++BIN_DIR = '/usr/bin'
+ 
+ 
+ def _program(name, args):
+diff -urNp ninja-1.11.1.1.orig/src/ninja/ninja_syntax.py ninja-1.11.1.1/src/ninja/ninja_syntax.py
+--- ninja-1.11.1.1.orig/src/ninja/ninja_syntax.py	1970-01-01 01:00:00.000000000 +0100
++++ ninja-1.11.1.1/src/ninja/ninja_syntax.py	2024-04-05 18:14:49.870032963 +0100
```

Comment:
What's going on here? It looks like the patch is adding a file that's empty or not present. Is it auto-generated or something?

Reply:
Read up to line 7! :)

Reply:
Write smaller PRs if we want me to read the whole thing! :-)